src/components/details/details.component.ts

```diff
@@ -65,11 +67,31 @@ export default class SlDetails extends ShoelaceElement {
   @property({ type: Boolean, reflect: true }) disabled = false;
 
   firstUpdated() {
-    this.body.hidden = !this.open;
     this.body.style.height = this.open ? 'auto' : '0';
+    if (this.open) {
+      this.details.open = true;
+    }
+
+    this.detailsObserver = new MutationObserver(changes => {
+      for (const change of changes) {
+        if (change.type === 'attributes' && change.attributeName === 'open') {
+          if (this.details.open) {
+            this.show();
+          } else {
+            this.hide();
+          }
+        }
+      }
+    });
+    this.detailsObserver.observe(this.details, { attributes: true });
+  }
```

Comment:
Clever! I didn't think of handling it this way in my previous attempts to use `<details>`.